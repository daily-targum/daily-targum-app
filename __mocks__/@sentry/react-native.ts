export const init = () => jest.fn();