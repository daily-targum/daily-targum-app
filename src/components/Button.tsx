import React from 'react';
import { TouchableOpacity, ViewStyle, Text, TextStyle } from 'react-native';
import { useStyleCreator, makeStyleCreator, useTheme } from './Theme';
import { styleHelpers } from '../utils';

export function Button({
  children,
  onPress,
  style
}: {
  children: string | React.ReactNode,
  onPress: () => any,
  style?: ViewStyle,
}) {
  const {colors} = useTheme();
  const styles = useStyleCreator(styleCreator);
  return (
    <TouchableOpacity 
      style={[styles.touchableHighlight, style]}
      onPress={onPress}
      activeOpacity={0.6}
    >
      {typeof children === 'string' ? (
        <Text style={styles.text}>{children}</Text>
      ) : (
        <>
          {children}
        </>
      )}
    </TouchableOpacity>
  );
}

function TextButton({
  children,
  onPress,
  style
}: {
  children: string,
  onPress: () => any,
  style?: TextStyle,
}) {
  const styles = useStyleCreator(styleCreator);
  return (
    <TouchableOpacity 
      style={[styles.touchableOpacity, style]}
      onPress={onPress}
    >
      <Text style={styles.textButton}>{children}</Text>
    </TouchableOpacity>
  );
}

const styleCreator = makeStyleCreator(theme => ({
  touchableHighlight: {
    ...styleHelpers.flex('row'),
    padding: theme.spacing(0.5),
    paddingLeft: theme.spacing(3),
    paddingRight: theme.spacing(3),
    alignItems: 'center',
    backgroundColor: theme.colors.button,
    borderRadius: theme.roundness()
  },
  touchableOpacity: {
    ...styleHelpers.flex('row'),
    alignItems: 'center'
  },
  text: {
    color: theme.colors.accent,
    textTransform: 'uppercase',
    fontWeight: '700',
    fontSize: 14
  },
  textButton: {
    color: theme.colors.accent,
    fontWeight: '500',
    fontSize: 17
  }
}));

Button.Text = TextButton;
export default Button;