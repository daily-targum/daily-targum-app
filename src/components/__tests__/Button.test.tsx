jest.mock('../Theme');

import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';
import { TouchableOpacity, Text } from 'react-native';

describe('<Button />', () => {

  it('button renders correctly', () => {
    const tree = renderer.create(
      <Button onPress={() => {}}>Click Me!</Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  it('button renders correctly with ReactNode child', () => {
    const tree = renderer.create(
      <Button onPress={() => {}}>
        <Text>Click Me!</Text>
      </Button>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('button onPress', () => {
    const onPress = jest.fn();
    const tree = renderer.create(
      <Button onPress={onPress}>Click Me!</Button>
    );
    tree.root.findByType(TouchableOpacity).props.onPress();
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('text button renders correctly', () => {
    const tree = renderer.create(
      <Button.Text onPress={() => {}}>Click Me!</Button.Text>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('text button onPress', () => {
    const onPress = jest.fn();
    const tree = renderer.create(
      <Button.Text onPress={onPress}>Click Me!</Button.Text>
    );
    tree.root.findByType(TouchableOpacity).props.onPress();
    expect(onPress).toHaveBeenCalledTimes(1);
  });

});