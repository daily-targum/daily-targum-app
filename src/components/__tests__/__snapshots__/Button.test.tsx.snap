// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> button renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "borderRadius": 10,
      "display": "flex",
      "flexDirection": "row",
      "opacity": 1,
      "padding": 3.5,
      "paddingLeft": 21,
      "paddingRight": 21,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#f70737",
        "fontSize": 14,
        "fontWeight": "700",
        "textTransform": "uppercase",
      }
    }
  >
    Click Me!
  </Text>
</View>
`;

exports[`<Button /> button renders correctly with ReactNode child 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eee",
      "borderRadius": 10,
      "display": "flex",
      "flexDirection": "row",
      "opacity": 1,
      "padding": 3.5,
      "paddingLeft": 21,
      "paddingRight": 21,
    }
  }
>
  <Text>
    Click Me!
  </Text>
</View>
`;

exports[`<Button /> text button renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#f70737",
        "fontSize": 17,
        "fontWeight": "500",
      }
    }
  >
    Click Me!
  </Text>
</View>
`;
